src/rest/rest_devices.c

```diff
@@ -306,48 +464,42 @@ int rest_devices_put_cb(const ulfius_req_t *req, ulfius_resp_t *resp, void *cont
         ulfius_set_empty_body_response(resp, 400);
         goto exit;
     }
-    if ((json_object_get(jdevice, "psk") == NULL) || (json_object_get(jdevice, "psk_id") == NULL))
-    {
-        ulfius_set_empty_body_response(resp, 400);
-        goto exit;
-    }
 
-    json_t *jstring = json_string(id);
-    json_object_set_new(jdevice, "uuid", jstring);
-
-    // if later stages fail, global list will be updated, but database file not
-    // consider updating list at the end
-    if (rest_devices_update_list(rest->devicesList, jdevice))
+    j_object = json_object_get(jdevice, "name");
+    if (j_object == NULL)
```

Comment:
This was the exact case a few commits back, but I changed it to this because `json_string_value()` returns an allocated buffer which later has to be freed while it's lifetime is rather short